system-addon/lib/HighlightsFeed.jsm

```diff
@@ -81,6 +90,20 @@ this.HighlightsFeed = class HighlightsFeed {
 
     SectionsManager.updateSection(SECTION_ID, {rows: this.highlights}, this.highlightsLastUpdated === 0 || broadcast);
     this.highlightsLastUpdated = Date.now();
+    this.imageCache.clear();
```

Comment:
What's the purpose of the cache if we clear it on every fetch? It seems like it would only be beneficial if multiple highlights ended up using the same image?

Reply:
The image cache cleared at the end of every fetchHighlights once it's been used (in that method). The async fetchImage calls then fill up the cache again for the next fetchHighlights call. I think, though I could be wrong, that none of the fetchImage calls are executed before fetchHighlights has finished executing, so they fill up the cache after it has been cleared. Without the image cache I found that subsequent refreshes caused flickering of images for existing cards as the src was unset then reset again asynchronously.

Reply:
Hrm.. That makes sense. Although it's also quite dangerous how it is right now. If something were to delay this `clear()` until after some images were fetched and set, then it's just throwing away that work.

The cache seems to be acting more as a prefetcher, but it could be a better actual cache if it only removed items that weren't used if the concern is holding on to fetched images too long. But this can be done in a followup bug.

At minimum better comment on why it's `clear`ing here and how the cache will still be useful.